runtime/src/main/java/org/amqphub/quarkus/qpid/jms/runtime/QpidJmsProducer.java

```diff
@@ -33,8 +33,8 @@ public class QpidJmsProducer {
     @Produces
     @ApplicationScoped
     @DefaultBean
-    public ConnectionFactory connectionFactory() {
-        return new JmsConnectionFactory(config.username.orElse(null), config.password.orElse(null), config.url);
+    public ConnectionFactory connectionFactory(ConnectionFactoryWrapper wrapper) {
+        return wrapper.wrap(new JmsConnectionFactory(config.username.orElse(null), config.password.orElse(null), config.url));
```

Comment:
In some ways it feels a bit weird that this always ends up calling 'wrap', even when it knew from the start it normally wont do anything as it was originally supplied an Optional that made it clear there was no original wrapper...is there any way to simplify this to avoid always wrapping even when not doing anything?

Reply:
I tried to use [InterceptionFactory](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#interception_factory) here but it does not support in Quarkus right now  because there are some technical issues, see
- https://github.com/quarkusio/quarkus/issues/3532

Also take a look at @mkouba https://github.com/quarkusio/quarkus/pull/29753#issuecomment-1408882117

I'm not very sure if there is some other ways and open for suggestions.

@gsmet @cescoffier any thought?

Reply:
I was more thinking, since we start out with an Optional having a wrapper in it, or more typically not having one, cant we somehow 'just pass that along' (as probably shows, I have no real experience with the CDI stuff and what is possible).

Reply:
The possible way is to use `Arc.container()` to get `ConnectionFactoryWrapper` at runtime.

Reply:
Since a boolean config is going to be added to decide whether to enable wrapping..might a check go on ConnectionFactoryWrapper, of both that config and whether the original Optional had any wrapper...then this bit can simply check whether any wrapping should occur, and not call wrap in the cases it knows it wont do anything (i.e when disabled by default, and when there was no actual wrapper) ?